wagtail/contrib/settings/templatetags/wagtailsettings_tags.py

```diff
@@ -12,10 +12,13 @@
 def get_settings(context, use_default_site=False):
     if use_default_site:
         site = Site.objects.get(is_default_site=True)
-        context['settings'] = SettingsProxy(site)
+        settings = SettingsProxy(site)
     elif 'request' in context:
-        context['settings'] = SettingsProxy(context['request'])
+        settings = SettingsProxy(context['request'])
     else:
         raise RuntimeError('No request found in context, and use_default_site '
                            'flag not set')
-    return ''
+    # For backward compatibility with the old behaviour of this tag
+    # set context['settings'] if it isn't already set
+    context.setdefault('settings', settings)
+    return settings
```

Comment:
Won't this cause the settings object to be rendered on the page, in the case that the old `{% get_settings %}` variant is used?

Reply:
Hah, yes, of course it would! I will rework this to use the approach you've suggested. Do you want to retain the current tag behaviour (writing to the context) indefinitely or should we add a deprecation warning if it is used without an assignment to a variable?

Reply:
I think we should continue to allow the current syntax without a deprecation warning - if someone's happily using the `settings` variable without clashes then there's no real harm in letting them continue doing that.